src/dsp_tools/commands/excel2json/project.py

```diff
@@ -41,14 +41,53 @@ def excel2json(
         True if everything went well
     """
 
-    overall_success = True
+    listfolder, onto_folders = _validate_folder_structure_get_filenames(data_model_files)
+
+    # create output
+    # -------------
+    overall_success, project = _create_project_json(data_model_files, listfolder, onto_folders)
+
+    with open(path_to_output_file, "w", encoding="utf-8") as f:
+        json.dump(project, f, indent=4, ensure_ascii=False)
+
+    print(f"JSON project file successfully saved at {path_to_output_file}")
+
+    return overall_success
 
-    # validate input
-    # --------------
+
+def _validate_folder_structure_get_filenames(data_model_files: str) -> tuple[list[Path], list[Path]]:
```

Comment:
Yes the functions are not great, but I did not want to change any code, just make it structurally easier, so that one function has minimum number of types of errors it generates.

The section we could take out, so this becomes little more than a function that collects functions, would be in my opinion the comprehension in line 61. What do you think?

Reply:
> The section we could take out, so this becomes little more than a function that collects functions, would be in my opinion the comprehension in line 61. What do you think?

I don't think that this would help. I would rather rename the function to `_get_filenames_from_folder()`. Because then, the name designates the actual, productive work that the function does (=getting files), and silences part that goes on behind the stage (=validating). But feel free to change it or leave it :-)

Reply:
I prefer to leave it, because it does not only get it but, also checks it and could raise an error.